Userland/Libraries/LibGfx/Painter.cpp

```diff
@@ -1874,40 +1853,51 @@ void Painter::draw_quadratic_bezier_curve(const IntPoint& control_point, const I
 // static
 void Painter::for_each_line_segment_on_elliptical_arc(const FloatPoint& p1, const FloatPoint& p2, const FloatPoint& center, const FloatPoint radii, float x_axis_rotation, float theta_1, float theta_delta, Function<void(const FloatPoint&, const FloatPoint&)>& callback)
 {
-    struct SegmentDescriptor {
-        FloatPoint p1;
-        FloatPoint p2;
-        float theta;
-        float theta_delta;
-    };
+    if (radii.x() == 0 || radii.y() == 0)
```

Comment:
Might as well ensure these are positive as well
```suggestion
    if (radii.x() <= 0 || radii.y() <= 0)
```

Reply:
Good point!